File: sdk/sdk-api-java/src/main/java/com/xpay/sdk/api/entity/Response.java
package com.xpay.sdk.api.entity;

import com.xpay.sdk.api.enums.RespCode;

/**
 * 响应给商户的VO
 */
public class Response {
    private String resp_code;
    private String resp_msg;
    private String data;
    private String mch_no;
    private String rand_str;
    private String sign;
    private String sign_type;
    private String sec_key;

    public String getResp_code() {
        return resp_code;
    }

    public void setResp_code(String resp_code) {
        this.resp_code = resp_code;
    }

    public String getResp_msg() {
        return resp_msg;
    }

    public void setResp_msg(String resp_msg) {
        this.resp_msg = resp_msg;
    }

    public String getData() {
        return data;
    }

    public void setData(String data) {
        this.data = data;
    }

    public String getMch_no() {
        return mch_no;
    }

    public void setMch_no(String mch_no) {
        this.mch_no = mch_no;
    }

    public String getRand_str() {
        return rand_str;
    }

    public void setRand_str(String rand_str) {
        this.rand_str = rand_str;
    }

    public String getSign() {
        return sign;
    }

    public void setSign(String sign) {
        this.sign = sign;
    }

    public String getSign_type() {
        return sign_type;
    }

    public void setSign_type(String sign_type) {
        this.sign_type = sign_type;
    }

    public String getSec_key() {
        return sec_key;
    }

    public void setSec_key(String sec_key) {
        this.sec_key = sec_key;
    }

    /**
     * 判断本次请求是否成功
     * @return
     */
    public boolean isSuccess(){
        return RespCode.SUCCESS.getCode().equals(resp_code);
    }

    /**
     * 获取resp_msg中的错误码
     * @return
     */
    public String getRespMsgCode(){
        int index = -1;
        if(resp_msg != null && (index = resp_msg.indexOf("[")) >= 0 && resp_msg.endsWith("]")){
            return resp_msg.substring(index+1, resp_msg.length()-1);
        }else{
            return null;
        }
    }
}

File: sdk/sdk-api-php7/entity/Response.php
<?php
namespace xpay;

/**
 * 响应参数类
 * Class Response
 * @package entity
 */
class Response {
    private $resp_code;
    private $resp_msg;
    private $mch_no;
    private $data;
    private $rand_str;
    private $sign_type;
    private $sign;
    private $sec_key;

    /**
     * @return mixed
     */
    public function getRespCode()
    {
        return $this->resp_code;
    }

    /**
     * @param mixed $resp_code
     */
    public function setRespCode($resp_code): void
    {
        $this->resp_code = $resp_code;
    }

    /**
     * @return mixed
     */
    public function getRespMsg()
    {
        return $this->resp_msg;
    }

    /**
     * @param mixed $resp_msg
     */
    public function setRespMsg($resp_msg): void
    {
        $this->resp_msg = $resp_msg;
    }

    /**
     * @return mixed
     */
    public function getMchNo()
    {
        return $this->mch_no;
    }

    /**
     * @param mixed $mch_no
     */
    public function setMchNo($mch_no): void
    {
        $this->mch_no = $mch_no;
    }

    /**
     * @return mixed
     */
    public function getData()
    {
        return $this->data;
    }

    /**
     * @param mixed $data
     */
    public function setData($data): void
    {
        $this->data = $data;
    }

    /**
     * @return mixed
     */
    public function getRandStr()
    {
        return $this->rand_str;
    }

    /**
     * @param mixed $rand_str
     */
    public function setRandStr($rand_str): void
    {
        $this->rand_str = $rand_str;
    }

    /**
     * @return mixed
     */
    public function getSignType()
    {
        return $this->sign_type;
    }

    /**
     * @param mixed $sign_type
     */
    public function setSignType($sign_type): void
    {
        $this->sign_type = $sign_type;
    }

    /**
     * @return mixed
     */
    public function getSign()
    {
        return $this->sign;
    }

    /**
     * @param mixed $sign
     */
    public function setSign($sign): void
    {
        $this->sign = $sign;
    }

    /**
     * @return mixed
     */
    public function getSecKey()
    {
        return $this->sec_key;
    }

    /**
     * @param mixed $sec_key
     */
    public function setSecKey($sec_key): void
    {
        $this->sec_key = $sec_key;
    }

    /**
     * 判断本次请求是否成功
     * @return bool
     */
    public function isSuccess(): bool
    {
        return "01" === $this->resp_code;
    }

    /**
     * 从resp_msg中获取错误码
     * @return string
     */
    public function getRespMsgCode(): string
    {
        if($this->resp_msg && ($index = strpos($this->resp_msg, "[")) >= 0
            && strpos($this->resp_msg, "]") == (strlen($this->resp_msg)-1)){
            $code = substr($this->resp_msg, $index+1);
            return substr($code, 0, strlen($code)-1);
        }else{
            return "";
        }
    }
}